etl/analytics/common.py

```diff
@@ -1,63 +1,315 @@
+    # 'span-link',
+    # 'topic-page-intro',
+    # 'video',
+]
+# Dictionary that maps the type of a gdoc post (as defined in the posts_gdocs DB table) to its base URL; adding the post's slug to this base URL gives the complete URL to the corresponding post.
+POST_TYPE_TO_URL = {
+    "about-page": OWID_BASE_URL,
+    "article": OWID_BASE_URL,
+    "author": f"{OWID_BASE_URL}team/",
+    "data-insight": f"{OWID_BASE_URL}data-insights/",
+    # Fragments are used just a handful of times, and seems to be used for data pages FAQ.
+    # It's not clear to me how to link them to specific posts, so, ignore them for now.
+    "fragment": None,
+    # Gdocs of type 'homepage' have an arbitrary slug "owid-homepage". They will be manually fixed later on.
+    "homepage": None,
+    "linear-topic-page": OWID_BASE_URL,
+    "topic-page": OWID_BASE_URL,
+}
+# Dictionary that maps a link type (as defined in the posts_gdocs_links DB table) to the base url; adding the target to this base url gives a full URL to the linked content (e.g. a grapher chart).
+POST_LINK_TYPES_TO_URL = {
+    "grapher": OWID_BASE_URL + "grapher/",
+    # Chart views, in theory, refer to narrative charts, which don't have a public URL.
+    # They are handled separately.
+    # NOTE: there are chart views for non-narrative charts, so there may be other cases I'm not considering.
+    "chart-view": OWID_BASE_URL + "grapher/",
+    "explorer": OWID_BASE_URL + "explorers/",
+    "gdoc": "https://docs.google.com/document/d/",
+    # A "url" link type links to an arbitrary URL.
+    # NOTE: In a handful of cases, there are links of type "url" and component type "chart". It's not clear to me why they are not of link types "grapher" with type "chart". But what's clear is that, when the link type is "grapher" the target corresponds to a slug, and when the link type is "url" the target is a full URL to a grapher chart. So, simply map these urls to their targets without modification.
+    "url": "",
+}
+
+
+def _try_to_execute_datasette_query(sql_url: str, warn: bool = False) -> pd.DataFrame:
+    try:
+        df = pd.read_csv(sql_url)
+        return df
+    except urllib.error.HTTPError as e:
+        if e.code == 414:
+            raise ValueError("HTTP 414: Query too long. Consider simplifying or batching the request.")
+        else:
+            raise
+
+
+def clean_sql_query(sql: str) -> str:
+    """
+    Normalize an SQL string for use in Datasette URL queries.
+
+    Parameters
+    ----------
+    sql : str
+        SQL query to clean.
+    Returns
+    -------
+    str
+        Cleaned SQL query.
 
-def read_datasette(sql: str, datasette_csv_url: str = ANALYTICS_CSV_URL) -> pd.DataFrame:
+    """
+    return " ".join(sql.strip().rstrip(";").split())
+
+
+def read_datasette(
```

Comment:
maybe worth having a dedicated module for all datasette-related tools? e.g. `analytics.datasette`